src/zenml/zen_stores/rest_zen_store.py

```diff
@@ -1354,7 +1360,7 @@ def get_run(
 
     def get_or_create_run(
         self, pipeline_run: PipelineRunRequestModel
-    ) -> PipelineRunResponseModel:
+    ) -> Tuple[PipelineRunResponseModel, bool]:
```

Comment:
I feel this breaks with a lot assumptions we have about how the rest zen store methods work. Why is it important here to not return the response model if it already exists

Reply:
We need to know whether it was created or not in `step_launcher.py`, because we only want to do the pipeline run metadata registration once per pipeline run.

Reply:
@schustmi and I discussed this for quite a while and, at the moment, having a separate `get_or_create` endpoint that gives us this bool is the best solution we could come up with. The alternatives would have been to 
- add `was_created_in_this_step` into the run model (which feels even more wrong) or
- have run metadata logging called multiple times per run (which would be unnecessary overhead and require race condition handling the back)

Reply:
Thanks, makes sense